cluster-operator/src/main/java/io/strimzi/operator/cluster/model/EntityTopicOperator.java

```diff
@@ -277,6 +280,7 @@ public static EntityTopicOperator fromCrd(Reconciliation reconciliation, Kafka k
         varList.add(buildEnvVar(ENV_VAR_FULL_RECONCILIATION_INTERVAL_MS, Integer.toString(reconciliationIntervalMs)));
         varList.add(buildEnvVar(ENV_VAR_ZOOKEEPER_SESSION_TIMEOUT_MS, Integer.toString(zookeeperSessionTimeoutMs)));
         varList.add(buildEnvVar(ENV_VAR_TOPIC_METADATA_MAX_ATTEMPTS, String.valueOf(topicMetadataMaxAttempts)));
+        varList.add(buildEnvVar(ENV_VAR_SECURITY_PROTOCOL, EntityTopicOperatorSpec.DEFAULT_SECURITY_PROTOCOL));
```

Comment:
Maybe I missed something. But don't we need to also set `STRIMZI_TLS_AUTH_ENABLED` here?

Reply:
I think you are somewhat correct (I've misconfigured it but should be fixed in latest commit). `STRIMZI_TLS_AUTH_ENABLED` and `STRIMZI_PUBLIC_CA` only gate if certificates should be generated, they're not used in the operator. The issue I ran into is that we need to explicitly set the generation of truststore and keystore to false otherwise the `tls_operator_run.sh` will generate them by checking if for example `STRIMZI_TRUSTORE_LOCATION` is undefined(`[ -z "$STRIMZI_TRUSTSTORE_LOCATION" ]` ) which always defaults to that when the string is empty. This conditional sets the `STRIMZI_TRUSTSTORE_LOCATION` and related keystore/truststore variables which we don't want with an external cluster. They're not used in the topic operator anymore, only in the sh scripts `topic_operator_run.sh`. 

They're used here:

https://github.com/strimzi/strimzi-kafka-operator/blob/bbef892e8f398098c45f23a9cfb0344ecf7cf924/topic-operator/scripts/topic_operator_run.sh#L27-L39 to avoid setting the keystore/truststore variables. And they skip setting/generating stores when the `PUBLIC_CA` is true and `TLS_AUTH_ENABLED` is false. In all other cases the checks should not affect anything. Let me know if there is a better/cleaner way.

In the operator we just check if the truststore/keystore locations/passwords are not empty: https://github.com/strimzi/strimzi-kafka-operator/pull/5201/files#diff-0ed74d5df887eaebfe8c8039af02c2bd34e16cf69e9a094bd0814a6879b58e94R325-R339